waka_waka/Scenes/maze.gd

```diff
@@ -0,0 +1,256 @@
+extends TileMapLayer
+
+@export var maze_width: int = 90
+@export var maze_height: int = 90
+
+const DIRECTIONS: Array = [
+	Vector2i(0, -1), # North
+	Vector2i(1, 0),  # East
+	Vector2i(0, 1),  # South
+	Vector2i(-1, 0)  # West
+]
+
+# Lookup table: maps hashed PackedByteArray (NESW) to (atlas_id, # of clockwise rotations)
```

Comment:
For readability, I would but these into a KeyValue to be looked up by name instead of a hash of the open edges. I assume that's relevant for the generative algorithm in some way?

You should create ROTATE_90 = Vector2i({whatever the numbers are) as a constant. There's no reason to recreate that object each time (as well as ROTATE_180, ROTATE_270)

I understand the ID being the "piece type" (wall/encapetc) but why is it a vector? What does this do?

Reply:
I agree with you re: using key values with better names, added to todo.

re: ID being the "piece type" => right now this maps directly to atlas_id, which is basically "Vector2i(col, row) on the sprite sheet", and they're defined as Vector2i, eg: https://docs.godotengine.org/en/stable/classes/class_tilemap.html#class-tilemap-method-set-cell

Will rename this key "atlas_id".